supernova/src/ccs/lccs.rs

```diff
@@ -0,0 +1,448 @@
+use ark_crypto_primitives::sponge::CryptographicSponge;
+use ark_ec::{AdditiveGroup, CurveGroup};
+use ark_ff::Field;
+use ark_poly::{
+    DenseMVPolynomial, DenseMultilinearExtension, MultilinearExtension, SparseMultilinearExtension,
+};
+use ark_poly_commit::{LabeledPolynomial, PolynomialCommitment};
+use ark_serialize::{CanonicalDeserialize, CanonicalSerialize};
+
+use ark_std::ops::Index;
+use std::ops::Neg;
+
+#[cfg(feature = "parallel")]
+use rayon::iter::{IntoParallelIterator, IntoParallelRefIterator, ParallelIterator};
+
+use super::super::r1cs::R1CSShape;
+pub use super::super::sparse::{MatrixRef, SparseMatrix};
+use super::mle::{fold_vec_to_mle_low, matrix_to_mle, mle_to_mvp, vec_to_mle};
+
+#[derive(Debug, Copy, Clone, PartialEq)]
+pub enum Error {
+    InvalidWitnessLength,
+    InvalidInputLength,
+    InvalidEvaluationPoint,
+    InvalidTargets,
+    FailedWitnessCommitting,
+    NotSatisfied,
+}
+
+pub struct LCCSShape<G: CurveGroup> {
```

Comment:
we might as well keep it if you've already written it, maybe inside this module's tests - it will likely be useful once we have non-r1cs constraints.